File: XTApp/XTApp/XTMy/XTView/XTMyHeadView.m
//
//  XTMyHeadView.m
//  XTApp
//
//  Created by xia on 2024/9/12.
//

#import "XTMyHeadView.h"
#import "XTMyModel.h"

@interface XTMyHeadView()

@property(nonatomic,strong) UILabel *tagLab;

@end

@implementation XTMyHeadView

- (instancetype)init{
    self = [super initWithFrame:CGRectMake(0, 0, XT_Screen_Width, 166 + XT_StatusBar_Height)];
    if(self){
        [self addSubview:[UIView xt_frame:CGRectMake(0, 0, self.width, XT_StatusBar_Height) color:XT_RGB(0x0BB559, 1.0f)]];

        self.backgroundColor = XT_RGB(0x0BB559, 1.0f);
        UIImageView *logoImg = [UIImageView xt_img:@"xt_my_head_logo" tag:0];
        [self addSubview:logoImg];
        [logoImg mas_makeConstraints:^(MASConstraintMaker *make) {
            make.top.equalTo(self.mas_top).offset(-9 + XT_StatusBar_Height);
            make.left.equalTo(self.mas_left).offset(18);
            make.size.mas_equalTo(CGSizeMake(91, 96));
        }];
        
        UILabel *phoneLab = [UILabel xt_lab:CGRectZero text:[XTUserManger xt_share].xt_user.xt_phone.xt_phonePrivacy font:XT_Font_SD(20) textColor:[UIColor whiteColor] alignment:NSTextAlignmentLeft tag:0];
        [self addSubview:phoneLab];
        [phoneLab mas_makeConstraints:^(MASConstraintMaker *make) {
            make.top.equalTo(self.mas_top).offset(18 + XT_StatusBar_Height);
            make.left.equalTo(logoImg.mas_right).offset(8);
            make.height.mas_equalTo(28);
        }];

        [self addSubview:self.tagLab];
        [self.tagLab mas_makeConstraints:^(MASConstraintMaker *make) {
            make.top.equalTo(phoneLab.mas_bottom).offset(7);
            make.left.equalTo(logoImg.mas_right).offset(8);
            make.height.mas_equalTo(21);
        }];
        
        UIImageView *footBgImg = [UIImageView xt_img:@"xt_my_foot_bg" tag:0];
        [self addSubview:footBgImg];
        [footBgImg mas_makeConstraints:^(MASConstraintMaker *make) {
            make.left.right.bottom.equalTo(self);
            make.height.mas_equalTo(80);
        }];
        
        UIImageView *iconImg = [UIImageView xt_img:@"xt_my_head_list_logo" tag:0];
        [self addSubview:iconImg];
        [iconImg mas_makeConstraints:^(MASConstraintMaker *make) {
            make.left.equalTo(self.mas_left).offset(35);
            make.top.equalTo(footBgImg.mas_top).offset(17);
            make.size.mas_equalTo(CGSizeMake(21, 21));
        }];
        
        UILabel *nameLab = [UILabel xt_lab:CGRectZero text:@"My list" font:XT_Font_M(18) textColor:XT_RGB(0x161616, 1.0f) alignment:NSTextAlignmentLeft tag:0];
        [self addSubview:nameLab];
        [nameLab mas_makeConstraints:^(MASConstraintMaker *make) {
            make.left.equalTo(iconImg.mas_right).offset(9);
            make.centerY.equalTo(iconImg);
            make.height.mas_equalTo(21);
        }];
        
        UILabel *subLab = [UILabel xt_lab:CGRectZero text:@"Click to view my order status" font:XT_Font(12) textColor:XT_RGB(0x888888, 1.0f) alignment:NSTextAlignmentLeft tag:0];
        [self addSubview:subLab];
        [subLab mas_makeConstraints:^(MASConstraintMaker *make) {
            make.left.equalTo(iconImg);
            make.top.equalTo(iconImg.mas_bottom).offset(4);
            make.height.mas_equalTo(17);
        }];
        
        UIButton *btn = [UIButton xt_btn:@"Click enter" font:XT_Font_M(14) textColor:XT_RGB(0x010000, 1.0f) cornerRadius:16 tag:0];
        [self addSubview:btn];
        [btn mas_makeConstraints:^(MASConstraintMaker *make) {
            make.right.equalTo(self.mas_right).offset(-28);
            make.top.equalTo(footBgImg.mas_top).offset(22);
            make.size.mas_equalTo(CGSizeMake(120, 32));
        }];
        [btn.layer insertSublayer:[UIView xt_layer:@[(__bridge id)XT_RGB(0x02CC56, 1.0f).CGColor,(__bridge id)XT_RGB(0xC6FF95, 1.0f).CGColor] locations:@[@0,@1.0f] startPoint:CGPointMake(0.54, 0.85) endPoint:CGPointMake(0.54, 0) size:CGSizeMake(120, 32)] atIndex:0];
        @weakify(self)
        btn.rac_command = [[RACCommand alloc]initWithSignalBlock:^RACSignal *(id input) {
            @strongify(self)
            if(self.block) {
                self.block();
            }
            return [RACSignal empty];
        }];
    }
    return self;
}

- (void)setModel:(XTMyModel *)model {
    _model = model;
    if([NSString xt_isEmpty:model.xt_memberUrl]) {
        self.tagLab.text = @"Hello, New User";
    }
    else {
        self.tagLab.text = @"Hello, Esteemed Member";
    }
    
}

- (UILabel *)tagLab {
    if(!_tagLab) {
        _tagLab = [UILabel xt_lab:CGRectZero text:@"Hello, Esteemed Member" font:XT_Font_M(15) textColor:[UIColor whiteColor] alignment:NSTextAlignmentLeft tag:0];
    }
    return _tagLab;
}

/*
// Only override drawRect: if you perform custom drawing.
// An empty implementation adversely affects performance during animation.
- (void)drawRect:(CGRect)rect {
    // Drawing code
}
*/

@end

File: XTApp/XTApp/XTMy/XTView/XTMyHeadView.swift
//
//  XTMyHeadView.swift
//  XTApp
//
//  Created by Codex on 2026/4/26.
//

import UIKit

private func xtHeadColor(_ rgbValue: Int, alpha: CGFloat = 1.0) -> UIColor {
    UIColor(
        red: CGFloat((rgbValue & 0xFF0000) >> 16) / 255.0,
        green: CGFloat((rgbValue & 0x00FF00) >> 8) / 255.0,
        blue: CGFloat(rgbValue & 0x0000FF) / 255.0,
        alpha: alpha
    )
}

private var xtHeadStatusHeight: CGFloat {
    UIApplication.shared.statusBarFrame.size.height
}

private func xtHeadLabel(text: String, font: UIFont, color: UIColor, alignment: NSTextAlignment = .left) -> UILabel {
    let label = UILabel(frame: .zero)
    label.text = text
    label.font = font
    label.textColor = color
    label.textAlignment = alignment
    return label
}

private func xtPhonePrivacy(_ phone: String?) -> String {
    guard let phone, phone.count >= 7 else {
        return phone ?? ""
    }
    let prefix = phone.prefix(3)
    let suffix = phone.suffix(3)
    return "\(prefix)\(String(repeating: "*", count: phone.count - 6))\(suffix)"
}

@objcMembers
@objc(XTMyHeadView)
class XTMyHeadView: UIView {
    weak dynamic var model: XTMyModel? {
        didSet {
            if model?.xt_memberUrl?.isEmpty ?? true {
                tagLab.text = "Hello, New User"
            } else {
                tagLab.text = "Hello, Esteemed Member"
            }
        }
    }

    dynamic var block: XTBlock?

    private lazy var tagLab = xtHeadLabel(
        text: "Hello, Esteemed Member",
        font: .systemFont(ofSize: 15, weight: .medium),
        color: .white
    )

    init() {
        super.init(frame: CGRect(x: 0, y: 0, width: UIScreen.main.bounds.width, height: 166 + xtHeadStatusHeight))
        setupUI()
    }

    override init(frame: CGRect) {
        super.init(frame: frame)
        setupUI()
    }

    required init?(coder: NSCoder) {
        super.init(coder: coder)
        setupUI()
    }

    private func setupUI() {
        backgroundColor = xtHeadColor(0x0BB559)

        let statusView = UIView(frame: CGRect(x: 0, y: 0, width: bounds.width, height: xtHeadStatusHeight))
        statusView.backgroundColor = xtHeadColor(0x0BB559)
        addSubview(statusView)

        let logoImg = UIImageView(image: UIImage(named: "xt_my_head_logo"))
        logoImg.translatesAutoresizingMaskIntoConstraints = false
        addSubview(logoImg)
        NSLayoutConstraint.activate([
            logoImg.topAnchor.constraint(equalTo: topAnchor, constant: -9 + xtHeadStatusHeight),
            logoImg.leftAnchor.constraint(equalTo: leftAnchor, constant: 18),
            logoImg.widthAnchor.constraint(equalToConstant: 91),
            logoImg.heightAnchor.constraint(equalToConstant: 96)
        ])

        let phoneLab = xtHeadLabel(
            text: xtPhonePrivacy(UserSession.shared.currentUser?.phone),
            font: .systemFont(ofSize: 20, weight: .semibold),
            color: .white
        )
        phoneLab.translatesAutoresizingMaskIntoConstraints = false
        addSubview(phoneLab)
        NSLayoutConstraint.activate([
            phoneLab.topAnchor.constraint(equalTo: topAnchor, constant: 18 + xtHeadStatusHeight),
            phoneLab.leftAnchor.constraint(equalTo: logoImg.rightAnchor, constant: 8),
            phoneLab.heightAnchor.constraint(equalToConstant: 28)
        ])

        tagLab.translatesAutoresizingMaskIntoConstraints = false
        addSubview(tagLab)
        NSLayoutConstraint.activate([
            tagLab.topAnchor.constraint(equalTo: phoneLab.bottomAnchor, constant: 7),
            tagLab.leftAnchor.constraint(equalTo: logoImg.rightAnchor, constant: 8),
            tagLab.heightAnchor.constraint(equalToConstant: 21)
        ])

        let footBgImg = UIImageView(image: UIImage(named: "xt_my_foot_bg"))
        footBgImg.translatesAutoresizingMaskIntoConstraints = false
        addSubview(footBgImg)
        NSLayoutConstraint.activate([
            footBgImg.leftAnchor.constraint(equalTo: leftAnchor),
            footBgImg.rightAnchor.constraint(equalTo: rightAnchor),
            footBgImg.bottomAnchor.constraint(equalTo: bottomAnchor),
            footBgImg.heightAnchor.constraint(equalToConstant: 80)
        ])

        let iconImg = UIImageView(image: UIImage(named: "xt_my_head_list_logo"))
        iconImg.translatesAutoresizingMaskIntoConstraints = false
        addSubview(iconImg)
        NSLayoutConstraint.activate([
            iconImg.leftAnchor.constraint(equalTo: leftAnchor, constant: 35),
            iconImg.topAnchor.constraint(equalTo: footBgImg.topAnchor, constant: 17),
            iconImg.widthAnchor.constraint(equalToConstant: 21),
            iconImg.heightAnchor.constraint(equalToConstant: 21)
        ])

        let nameLab = xtHeadLabel(
            text: "My list",
            font: .systemFont(ofSize: 18, weight: .medium),
            color: xtHeadColor(0x161616)
        )
        nameLab.translatesAutoresizingMaskIntoConstraints = false
        addSubview(nameLab)
        NSLayoutConstraint.activate([
            nameLab.leftAnchor.constraint(equalTo: iconImg.rightAnchor, constant: 9),
            nameLab.centerYAnchor.constraint(equalTo: iconImg.centerYAnchor),
            nameLab.heightAnchor.constraint(equalToConstant: 21)
        ])

        let subLab = xtHeadLabel(
            text: "Click to view my order status",
            font: .systemFont(ofSize: 12),
            color: xtHeadColor(0x888888)
        )
        subLab.translatesAutoresizingMaskIntoConstraints = false
        addSubview(subLab)
        NSLayoutConstraint.activate([
            subLab.leftAnchor.constraint(equalTo: iconImg.leftAnchor),
            subLab.topAnchor.constraint(equalTo: iconImg.bottomAnchor, constant: 4),
            subLab.heightAnchor.constraint(equalToConstant: 17)
        ])

        let btn = UIButton(type: .custom)
        btn.setTitle("Click enter", for: .normal)
        btn.titleLabel?.font = .systemFont(ofSize: 14, weight: .medium)
        btn.setTitleColor(xtHeadColor(0x010000), for: .normal)
        btn.layer.cornerRadius = 16
        btn.clipsToBounds = true
        btn.addTarget(self, action: #selector(tapEnter), for: .touchUpInside)
        btn.translatesAutoresizingMaskIntoConstraints = false
        addSubview(btn)
        NSLayoutConstraint.activate([
            btn.rightAnchor.constraint(equalTo: rightAnchor, constant: -28),
            btn.topAnchor.constraint(equalTo: footBgImg.topAnchor, constant: 22),
            btn.widthAnchor.constraint(equalToConstant: 120),
            btn.heightAnchor.constraint(equalToConstant: 32)
        ])

        let gradient = CAGradientLayer()
        gradient.colors = [xtHeadColor(0x02CC56).cgColor, xtHeadColor(0xC6FF95).cgColor]
        gradient.locations = [0, 1.0]
        gradient.startPoint = CGPoint(x: 0.54, y: 0.85)
        gradient.endPoint = CGPoint(x: 0.54, y: 0)
        gradient.frame = CGRect(x: 0, y: 0, width: 120, height: 32)
        btn.layer.insertSublayer(gradient, at: 0)
    }

    @objc private func tapEnter() {
        block?()
    }
}
